pymc3/distributions/discrete.py

```diff
@@ -1335,18 +1302,22 @@ def logp(self, value):
         -------
         TensorVariable
         """
-        psi = self.psi
-        theta = self.theta
 
         logp_val = at.switch(
             at.gt(value, 0),
-            at.log(psi) + self.pois.logp(value),
+            at.log(psi) + Poisson.logp(value, theta),
```

Comment:
Those `logp` functions inside the `Distribution` classes are just a convenience for declaring dispatch functions, so we don't want to rely on them in any way.  Instead, call the underlying `_logp` directly.